### DISCUSSION TITLE
Detect user vs plugin file edits in VS Code extension

### DISCUSSION THREAD

**Discussion by sayroll** [microsoft/vscode-discussions#1157] [Extension Development QnA] [answered]
Hello everyone,

I'm facing an issue during my plugin development process and I need some insights. I noticed that both manual edits performed by users in the VS Code window and edits made using WorkspaceEdit trigger the onDidChangeTextDocument event. However, I have a requirement to differentiate between events caused by manual user interactions and those caused by edits made using WorkspaceEdit in the onDidChangeTextDocument event.

After searching through available resources, I haven't been able to find a solution to this problem. If anyone has any insights or suggestions, I would greatly appreciate it. Thank you so much!

**Comment by alefragnani** [comment, marked as answer]
Hi @sayroll ,

You are right, the current API doesn't expose the _source_ of the text change. Looking at VS Code repo issues, I could find the one that exposed the `reason: TextDocumentChangeReason` attribute (https://github.com/microsoft/vscode/issues/120617). Originally, it requested what you asked, but in the end, only `undo` and `redo` was made available.

I don't see any other open request about new `reasons`, so I would suggest you to open a feature request at VS Code repo (https://github.com/microsoft/vscode/issues/new/choose), asking for new `reasons`, like originally requested on that issue.

Hope this helps

**Reply by sayroll** [reply]
Got it. I will open a feature request at the VS Code repository. Thank you so much for your response.
